src/main/java/com/cgi/ipsen5/Repository/ReservationHistoryRepository.java

```diff
@@ -27,13 +27,16 @@ List<RoomOccupancyResponseDTO> findRoomOccupancyByBuildingAndYear(
     @Query(value = "SELECT CONCAT(e.first_name, ' ', e.last_name) AS employeeName, COUNT(*) AS numberOfReservations, " +
             "SUM(CASE WHEN rh.status = 'NOT_CHECKED_IN' THEN 1 ELSE 0 END) AS numberOfNoShows " +
             "FROM reservation_history rh " +
-            "JOIN employee e ON rh.user_id = e.id " +
-            "JOIN location l ON rh.location_id = l.id " +
-            "JOIN wing w ON l.wing_id = w.id " +
-            "JOIN floor f ON w.floor_id = f.id " +
-            "JOIN building b ON f.building_id = b.id " +
+            "INNER JOIN employee e ON rh.user_id = e.id " +
+            "INNER JOIN location l ON rh.location_id = l.id " +
+            "INNER JOIN wing w ON l.wing_id = w.id " +
+            "INNER JOIN floor f ON w.floor_id = f.id " +
+            "INNER JOIN building b ON f.building_id = b.id " +
             "WHERE b.id = :building AND EXTRACT(YEAR FROM rh.start_date_time) = :year " +
-            "GROUP BY employeeName", nativeQuery = true)
+            "GROUP BY employeeName " +
+            "HAVING SUM(CASE WHEN rh.status = 'NOT_CHECKED_IN' THEN 1 ELSE 0 END) > 0 " +
+            "ORDER BY numberOfNoShows DESC " +
+            "LIMIT 5", nativeQuery = true)
```

Comment:
Wil je dit limiteren tot 5? Misschien beter in de frontend afhandelen?

Reply:
Volgens mij had Leo een keer in een interview benoemd dat hij de werknemers in een tabel wilde zien met de meeste no shows en het lijkt mij beter om dit gelijk mee te filteren in de backend dan frontend.